src/Core/Webservice/WebserviceCanBeEnabledConfigurationChecker.php

```diff
@@ -0,0 +1,171 @@
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Webservice;
+
+use PrestaShop\PrestaShop\Adapter\Configuration;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Translation\TranslatorInterface;
+use RuntimeException;
+
+/**
+ * Looks at server configuration in order to check PrestaShop Webservice can be enabled
+ */
+class WebserviceCanBeEnabledConfigurationChecker
+{
+    const ISSUE_NOT_APACHE_SERVER = 'not_apache_server';
+    const ISSUE_CANNOT_CHECK_APACHE_MODULES = 'cannot_check_apache_modules';
+    const ISSUE_APACHE_MOD_AUTH_BASIC_NOT_AVAILABLE = 'issue_apache_mod_auth_basic_not_available';
+    const ISSUE_APACHE_MOD_AUTH_REWRITE_NOT_AVAILABLE = 'issue_apache_mod_auth_rewrite_not_available';
+    const ISSUE_EXT_SIMPLEXML_NOT_AVAILABLE = 'issue_ext_simplexml_not_available';
+    const ISSUE_HTTPS_NOT_AVAILABLE = 'issue_https_not_available';
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var Configuration
+     */
+    private $configuration;
+
+    /**
+     * @param TranslatorInterface $translator
+     * @param Configuration $configuration
+     */
+    public function __construct(TranslatorInterface $translator, Configuration $configuration)
+    {
+        $this->translator = $translator;
+        $this->configuration = $configuration;
+    }
+
+    /**
+     * Analyses the server configuration (apache configuration and php settings)
+     * to check whether PrestaShop Webservice can be used or not.
+     *
+     * @param Request $request optional
+     *
+     * @return array empty if no errors
+     */
+    public function getErrors(Request $request = null)
+    {
+        $issues = $this->lookForIssues($request);
+
+        $allWarningMessages = $this->getWarningMessages();
+        $selectedWarningMessages = [];
+
+        foreach ($issues as $issue) {
+            if (false === array_key_exists($issue, $allWarningMessages)) {
+                throw new RuntimeException(sprintf('Unexpected configuration issue %s', $issue));
+            }
+
+            $selectedWarningMessages[] = $allWarningMessages[$issue];
+        }
+
+        return $selectedWarningMessages;
+    }
+
+    /**
+     * @param Request $request optional
+     *
+     * @return string[]
+     */
+    private function lookForIssues(Request $request = null)
+    {
+        $issues = [];
+
+        if ($request !== null) {
+            if (strpos($request->server->get('SERVER_SOFTWARE'), 'Apache') === false) {
+                $issues[] = self::ISSUE_NOT_APACHE_SERVER;
```

Comment:
Don't understand why using Nginx can be an issue?

Reply:
This is a rule that was applied in the legacy controller. I do not have the history.